File: examples/load.py
#!/usr/bin/env python3

import sys
from python_io import *


if __name__ == '__main__':
    npy_file = sys.argv[1]
    npz_file = sys.argv[2]

    A = load(npy_file)
    print_matrix(A)

    data = load(npz_file)
    print_dict(data)

File: examples/load.cpp
#include <cstdlib>
#include <Eigen/Dense>
#include "io.h"

int main(int argc, char* argv[])
{
  std::string npy_file = argv[1];
  std::string npz_file = argv[2];

  Eigen::MatrixXf A = load_npy(npy_file);
  print_matrix(A);

  std::map<std::string, Eigen::MatrixXf> data = load_npz(npz_file);
  print_dict(data);

  return EXIT_SUCCESS;
}
